### PR TITLE
Clarify cryptic comments

### PR DESCRIPTION
**Pull request by ouz-a**
Clarifies some unclear comments that lurked in the compiler.

r? @oli-obk

### CODE REVIEW

File: tests/ui/drop-bounds/drop-bounds-impl-drop.rs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 #![deny(drop_bounds)]
 // As a special exemption, `impl Drop` in the return position raises no error.
 // This allows a convenient way to return an unnamed drop guard.
+// Voldemort here refers to types that are unnameable.
 fn voldemort_type() -> impl Drop {
   struct Voldemort;
   impl Drop for Voldemort {
```
Comment (compiler-errors) [inline tests/ui/drop-bounds/drop-bounds-impl-drop.rs:8]:
```suggestion
// As a special exemption, `impl Drop` in the return position raises no error.
// This allows a convenient way to return an unnamed drop guard.
fn unnameable_type() -> impl Drop {
  struct Unnameable;
  impl Drop for Unnameable {
```

Comment (compiler-errors) [reply tests/ui/drop-bounds/drop-bounds-impl-drop.rs:8]: (and the call right below, but i can't suggest changes on that part of the file :smile_cat:)

File: tests/ui/drop-bounds/drop-bounds-impl-drop.rs [hunk 2]
```diff
@@ -2,13 +2,13 @@
...
+  struct Unnameable;
+  impl Drop for Unnameable {
     fn drop(&mut self) {}
   }
-  Voldemort
+  Unnameable
 }
 fn main() {
-  let _ = voldemort_type();
+  let _ = unnameable();
```
Comment (oli-obk) [inline tests/ui/drop-bounds/drop-bounds-impl-drop.rs:13]:
```suggestion
  let _ = unnameable_type();
```

**Pull request merged**